core/domain/opportunity_services.py

```diff
@@ -572,6 +572,31 @@ def get_exploration_opportunity_summaries_by_ids(ids):
     return opportunities
 
 
+def get_exploration_opportunity_summaries_by_topic_id(topic_id):
+    """Returns a list of ExplorationOpportunitySummary
+    with a given topic id.
+
+    Args:
+        topic_id: str. The topic for which opportunity summaries
+            are fetched.
+
+    Returns:
+        list(ExplorationOpportunitySummary). A list of
```

Comment:
Yes, this makes sense. I have been slightly confused since in some of the case cases, while explaining the input arg. in detail, it essentially turns out to be a repetition of the description. That is why below, I let the description be crisp and explained in detail below the description (as part of the input arg).